File: frontend/src/lib/api/history.ts
import { apiFetch } from "@/lib/api-client";

export type RunSummary = {
  run_id: string;
  started_at: string | null;
  completed_at: string | null;
  dry_run: boolean | null;
  sheet_names: string[] | null;
  routed_receipt_payment: number | null;
  routed_deposit_withdrawal: number | null;
  needs_review: number | null;
  duplicates_skipped: number | null;
};

export type LogEntry = {
  id: string;
  run_id: string;
  level: string;
  message: string;
  context: Record<string, unknown> | null;
  created_at: string;
};

export type StatsSummary = {
  total_processed: number;
  total_receipt_payment: number;
  total_deposit_withdrawal: number;
};

export function getRuns(limit = 20): Promise<RunSummary[]> {
  return apiFetch<RunSummary[]>(`/runs?limit=${limit}`);
}

export function getLogs(limit = 100): Promise<LogEntry[]> {
  return apiFetch<LogEntry[]>(`/logs?limit=${limit}`);
}

export function getStats(): Promise<StatsSummary> {
  return apiFetch<StatsSummary>("/stats");
}

File: backend/app/schemas/history.py
from datetime import datetime

from pydantic import BaseModel


class RunSummary(BaseModel):
    run_id: str
    started_at: datetime | None = None
    completed_at: datetime | None = None
    dry_run: bool | None = None
    sheet_names: list[str] | None = None
    routed_receipt_payment: int | None = None
    routed_deposit_withdrawal: int | None = None
    needs_review: int | None = None
    duplicates_skipped: int | None = None
    skipped_internal_credit: int | None = None


class LogEntry(BaseModel):
    id: str
    run_id: str
    level: str
    message: str
    context: dict | None = None
    created_at: datetime


class StatsSummary(BaseModel):
    total_processed: int
    total_receipt_payment: int
    total_deposit_withdrawal: int
